File: src/BukkitPE/event/block/BlockEvent.php
<?php
/**
 * Block related events
 */
namespace BukkitPE\event\block;

use BukkitPE\block\Block;
use BukkitPE\event\Event;

abstract class BlockEvent extends Event{
	/** @var \BukkitPE\block\Block */
	protected $block;

	/**
	 * @param Block $block
	 */
	public function __construct(Block $block){
		$this->block = $block;
	}

	/**
	 * @return Block
	 */
	public function getBlock(){
		return $this->block;
	}
}

File: src/main/java/net/BukkitPE/event/block/BlockEvent.java
package net.BukkitPE.event.block;

import net.BukkitPE.block.Block;
import net.BukkitPE.event.Event;

/**
 * BukkitPE Project
 */
public abstract class BlockEvent extends Event {

    protected final Block block;

    public BlockEvent(Block block) {
        this.block = block;
    }

    public Block getBlock() {
        return block;
    }
}
